### PR TITLE
Fix parsing usage warnings from codetools without location info

### LINKED ISSUE #1917: object_usage_linter fails to parse inline function output
```r
lintr::lint("
foo <- function() s
", lintr::object_usage_linter())
```

Throws a warning about an internal error:

```
Warning message:
In parse_check_usage(fun, known_used_symbols = known_used_symbols,  :
  Possible bug in lintr: Couldn't parse usage message ‘<anonymous>: no visible binding for global variable 's'
’. Ignoring 1 usage warnings. Please report an issue at https://github.com/r-lib/
```

### LINKED ISSUE #1986: New warning in 'designer' from object_usage_linter
Haven't confirmed this is a regression yet, but we observe this warning on main that is not present on v3.0.2:

```
Possible bug in lintr: Couldn't parse usage message ‘<anonymous>: no visible global function definition for 'find_cache_dir'
’. Ignoring 1 usage warnings. Please report an issue at https://github.com/r-lib/lintr/issues.
```

### PR DESCRIPTION
**Pull request by AshesITR**
Fixes #1986, Fixes #1917

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -22,8 +22,8 @@
 * `assignment_linter()` no longer lints assignments in braces that include comments when `allow_trailing = FALSE` (#1701, @ashbaldry)
 
 * `object_usage_linter()`
-   + No longer silently ignores usage warnings that don't contain a quoted name (#1714, @AshesITR)
-   + No longer fails on code with comments inside a multi-line call to `glue::glue()` (#1919, @MichaelChirico)
+  + No longer silently ignores usage warnings that don't contain a quoted name (#1714, @AshesITR)
```
Comment (AshesITR) [inline NEWS.md:25]: Indentation was inconsistent here (3 spaces instead of 2 spaces)

**Comment by AshesITR** [comment]
There is still one ugly edge case, but I fear it will be very hard to properly fix:

```r
codetools::checkUsage({
  foo <- function() a <-
    bar()
})
# This one will fail to find col1 because the symbol is on line 2:
# <anonymous>: no visible global function definition for ‘bar’
# This one is fixed by the PR:
# <anonymous>: local variable ‘a’ assigned but may not be used
```

File: tests/testthat/test-object_usage_linter.R [hunk 2]
```diff
@@ -765,3 +765,42 @@ test_that("functional lambda definitions are also caught", {
     object_usage_linter()
   )
 })
+
+test_that("messages without location info are repaired", {
```
Comment (MichaelChirico) [inline tests/testthat/test-object_usage_linter.R:769]: one more test to add if you don't mind:

when the object usage comes from the formals

```r
foo <- function(a = bar()) {
  a
}
```

**Review by MichaelChirico — COMMENTED** [review]
nice! lazy matching... my Achilles heel!

it looks like this may have popped new lints in quotes_linter()?

**Comment by AshesITR** [comment]
Interesting.

Also seeing a warning in lint-changed-files:

Linter consecutive_stopifnot_linter was deprecated in lintr version 3.1.0. Use consecutive_assertion_linter instead.

**Comment by AshesITR** [comment]
Thinking about the lint in R/quotes_linter.R actually helped me find a solution for the multi-line edge case I presented above 🥳

**Comment by AshesITR** [comment]
also fixes  #1917

File: NEWS.md [hunk 3]
```diff
@@ -41,7 +41,9 @@
 
 * `object_name_linter()` allows all S3 group Generics (see `?base::groupGeneric`) and S3 generics defined in a different file in the same package (#1808, #1841, @AshesITR)
 
-* `object_usage_linter()` improves identification of the exact source of a lint for undefined variables in expressions with where the variable is used as a symbol in a usual way, for example in a formula or in an extraction with `$` (#1914, @MichaelChirico).
+* `object_usage_linter()` improves identification of the exact source of a lint
+  + for undefined variables in expressions with where the variable is used as a symbol in a usual way, for example in a formula or in an extraction with `$` (#1914, @MichaelChirico).
```
Comment (MichaelChirico) [inline NEWS.md:45]: Cite #1917 as well? (and include `Fixes #1917` in the issue body for it to auto-close)

**Review by MichaelChirico — APPROVED** [review]

**Pull request merged**